File: CSE19163 SLL .py
class SLList():
    class node:
        def __init__(self,data):
            self.element = data
            self.next = None

 

    def __init__(self):
        self.head = self.node(None)
        self.sz = 0
        
    def insertFirst(self,data):
        temp = self.node(data)
        
        if(self.head.next==None):
            self.head.element = data;
        else:
            temp.next = self.head
            self.head = temp

 

        return
    
    def insertLast(self,data):
        
        node = self.node(data)
        if(self.head.next == None):
            self.head = node
        else:
            curr = self.head
            while(curr.next!=None):
                curr = curr.next
            curr.next = node
        
        return
    
    def deleteFirst(self):
        node = self.head
        self.head = self.head.next
        del node
        
        return

 

    def deleteLast(self):
        temp = self.head
        while(temp):
            prev = temp
            temp = temp.next
        prev.next = None   
        del temp
        return
    
    def isEmpty(self):

 

        if(self.head.element == None): return True
        return False
        
    def listSize(self):
        
        node = self.head;
        if(self.isEmpty()): return 0
        self.sz = 1
        
        while node.next!=None:
            self.sz+=1
            node = node.next
        return self.sz
    
    def printList(self):
        if (self.isEmpty()):
            print("List is Empty")
        else:
            currentNode = self.head
            while(currentNode != None):
                print(currentNode.element, end =" ")
                currentNode = currentNode.next
            print("")

 

def testSLL():
    sll = SLList()
    inputs=int(input())
    while inputs>0:
        command=input()
        operation=command.split()
        if(operation[0]=="S"):
            print(sll.listSize())
        elif(operation[0]=="I"):
            print(sll.isEmpty())
        elif(operation[0]=="IF"):
            sll.insertFirst(int(operation[1]))
            sll.printList()
        elif(operation[0]=="IL"):
            sll.insertLast(int(operation[1]))
            sll.printList()
        elif(operation[0]=="DF"):
            sll.deleteFirst()
            sll.printList()
        elif(operation[0]=="DL"):
            sll.deleteLast()
            sll.printList()
        inputs-=1

 

def main():
    testSLL()

 

if __name__ == '__main__':
    main()

File: CSE19163 SLL .cpp
#include <iostream>
#include<string>
using namespace std;

template<class T>
class SLList {
protected:
	class Node {
	public:
		T element;
		Node *next;
		Node(T x) {
		    //@start-editable@

         this->element = x;			
			
			//@end-editable@
		}
	};
	Node *head;
	Node *tail;
	int n;

public:
	SLList() {
		n = 0;
		head = tail = NULL;
	}

	bool isEmpty(){
		//@start-editable@

    if(head ==NULL)
    {
        return  true;
    }
    return false;
    
		
		
		
		//@end-editable@
	}

	int size() {
	    //@start-editable@

      
        return n;
	    
	    //@end-editable@
	}

	bool insertLast(T x) {
	    //@start-editable@

         Node *temp = new Node(x);
      temp->element=x;
      temp->next=NULL;
      if(head==NULL)
      {
        head=temp;
        tail=temp;
        temp=NULL;
      }
      else
      {	
        tail->next=temp;
        tail=temp;
      }	
	n++;	
		
		//@end-editable@
		return true;
	}

	T insertFirst(T x) {
	    //@start-editable@

    Node *new_node = new Node(x);
     if (n==0)
    {
        head = new_node;
        tail = head;
    }
      else
      {
        new_node->next = head;
        head = new_node;
      }
    n++;
        
  
		
		
		
		//@end-editable@
		return x;
	}

	T removeLast() {
	    //@start-editable@

   
		if(n==0){
	        return NULL;
	    }
	       
	       Node *curr = head;
           Node *prev;
           curr = head;
           while(curr->next!=NULL)
         {
          prev = curr;
          curr = curr->next;	
         }
         tail=prev;
         prev->next=NULL;
         delete curr;
           T x;
           n--;
	    if(1==1)
		{
		
		         
			//@end-editable@
			return x;
		}
		
	}

	T removeFirst() {
	    //@start-editable@

    
	    
	    
	        Node* temp = head; 
            head = head->next; 
            delete temp;
            n--;
	        
	    
		
		
		 T x;
		
		//@end-editable@
		return x;
	}
//Prints the list
	void printlist(){
		if (!isEmpty()) {
			Node *temp = head;
			while (temp != NULL) {
				cout << temp->element << " ";
				temp = temp->next;
			}
			cout << endl;
			return;
		}
		cout << ("List is empty!")<<endl;
	}

};

//Driver Code
int main(){
	SLList<int> slist1;
 	string noOfInputs,str;
 	getline(cin, noOfInputs);
 	for(int i=0;i<stoi(noOfInputs);i++){
 	    getline(cin, str); 
 	    
 	    if (str.substr(0, 2) == "IF"){
 	        int value = stoi(str.substr(3, 5));
 	        slist1.insertFirst(value);
 	        slist1.printlist();
 	    }
 	    else if (str.substr(0, 2) == "IL"){
 	        int value = stoi(str.substr(3, 5));
 	        slist1.insertLast(value);
 	        slist1.printlist();
 	    }
 	    else if (str.substr(0, 2) == "DF"){
 	        slist1.removeFirst();
 	        slist1.printlist();
 	    }
 	    else if (str.substr(0, 2) == "DL"){
 	        slist1.removeLast();
 	        slist1.printlist();
 	    }
 	    else if (str.substr(0, 1) == "S"){
 	       cout<< slist1.size()<<endl;
 	    }
 	    else if (str.substr(0,1) == "I"){
 	        //cout<<slist1.isEmpty()<<endl;
 	        if(slist1.isEmpty()){
 	            cout<<"True"<<endl;
 	        }
 	        else{
 	            cout<<"False"<<endl;
 	        }
 	    }
 	}
}